### DISCUSSION TITLE
Custom loss functions and printout help

### DISCUSSION THREAD

**Discussion by ccclalala123** [MilesCranmer/PySR#723] [Q&A]
I want to find three expressions for $f,g,z,h$ in the following equation, where $T_1,T_2,T_3,T_4,b$ are known 3*3 matrices：

$f(\alpha_1,\alpha_2,\alpha_3)*T_1+g(\alpha_1,\alpha_2,\alpha_3)*T_2+z(\alpha_1,\alpha_2,\alpha_3)*T_3+h(\alpha_1,\alpha_2,\alpha_3)*T_4=b$

(where the inputs $\alpha_1,\alpha_2,\alpha_3$ are one-dimensional and the outputs $b$ are multiple 3*3 matrices through operation)

How to print $f,g,z,h$ expressions?  ($f,g,z,h$ are respectively $prediction1,prediction2,prediction3,prediction4$ in the following code)

I refer to this section : https://github.com/MilesCranmer/SymbolicRegression.jl/pull/143#issuecomment-1279205103

Here's my loss function code:
```
objective = """
function my_loss_function(tree, dataset::Dataset{T,L}, options)::L where {T,L}

    prediction1, complete = eval_tree_array(tree, dataset.X, options)
    if !complete
        return L(Inf)
    end

    prediction2, complete = eval_tree_array(tree, dataset.X, options)
    if !complete
        return L(Inf)
    end

    prediction3, complete = eval_tree_array(tree, dataset.X, options)
    if !complete
        return L(Inf)
    end

    prediction4, complete = eval_tree_array(tree, dataset.X, options)
    if !complete
        return L(Inf)
    end

    pre_jz = prediction1 .* T_1_jl .+ prediction2 .* T_2_jl .+ prediction3 .* T_3_jl.+ prediction4 .* T_4_jl

    diffs = pre_jz .- dataset.y
    
    loss = (sum(diffs .^ 2)) / (length(diffs))
    
    return loss

end
"""

model = PySRRegressor(
    populations=20,
    population_size=60,
    niterations=130,
    binary_operators=["+", "-", "*", "/"],
    loss_function=objective,
    warm_start=True

)
model.fit(X, y)
model.get_best().equation
```
In addition, I also tried the following method, but it will return an infinite loss:
```
objective = """
function my_loss_function(tree, dataset::Dataset{T,L}, options)::L where {T,L}

   tree.degree != 4 && return L(Inf)
    P_1 = tree.l.l
    Q_2 = tree.l.r
    Z_3 = tree.r.l
    T_4 = tree.r.r

    prediction1, complete = eval_tree_array(P_1, dataset.X, options)
    if !complete
        return L(Inf)
    end

    prediction2, complete = eval_tree_array(Q_2, dataset.X, options)
    if !complete
        return L(Inf)
    end

    prediction3, complete = eval_tree_array(Z_3, dataset.X, options)
    if !complete
        return L(Inf)
    end

    prediction4, complete = eval_tree_array(Z_4, dataset.X, options)
    if !complete
        return L(Inf)
    end

    pre_jz = prediction1 .* T_1_jl .+ prediction2 .* T_2_jl .+ prediction3 .* T_3_jl .+ prediction4 .* T_4_jl

    diffs = pre_jz .- dataset.y
    
    loss = (sum(diffs .^ 2)) / (length(diffs))
    
    return loss

end
"""
Any help would be greatly appreciated!

**Comment by Ansy9378** [comment]
Hello @ccclalala123 
I'm working on custom loss functions and , Have you figured out how to print out the predictions from loss function code
